src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -297,6 +298,111 @@ public AgentSettings LoadSettings()
             }
         }
 
+        public async Task UnconfigureAsync(CommandSettings command)
+        {
+            //stop, uninstall service and remove service config file
+            _term.Write(StringUtil.Loc("UninstallingService"));
+            if (_store.IsServiceConfigured())
+            {
+                var serviceControlManager = HostContext.GetService<IServiceControlManager>();
+                try
+                {
+                    serviceControlManager.UnconfigureService();
+                    _term.WriteLine(StringUtil.Loc("Success"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteLine(StringUtil.Loc("Failed"));
+                    throw;
+                }
+            }
+            else
+            {
+                _term.WriteLine(StringUtil.Loc("NotFound"));
```

Comment:
While testing on Unix, I noticed that the "svc,sh" prints many lines on the console and they appear between "Uninstalling service: " and "Success" message. I will have to change the strings to accommodate that: first will be WriteLine("Uninstalling service...") and then WriteLine("Service uninstalled successfully.")